core/services/relay/delegate.go

```diff
@@ -202,6 +203,29 @@ func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (types
 			ContractID:    spec.ContractID,
 			TransmitterID: spec.TransmitterID.String,
 		})
+	case types.Dydx:
+		r, exists := d.relayers[types.Dydx]
+		if !exists {
+			return nil, errors.New("no Dydx relay found; is Dydx enabled?")
+		}
+
+		var config dydx.RelayConfig
+		err := json.Unmarshal(spec.RelayConfig.Bytes(), &config)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'spec.RelayConfig' unmarshal")
+		}
+
+		if !spec.IsBootstrapPeer {
+			if !spec.TransmitterID.Valid {
+				return nil, errors.New("transmitterID is required for non-bootstrap jobs")
+			}
+		}
```

Comment:
Looks like we repeat this check a few times, worth factoring out?

Reply:
This IsBootstrapPeer seems to be used for OCR1 only, I believe.
Is it fine to just remove this code, when I know this is only supported for Ocr2?

Reply:
> This IsBootstrapPeer seems to be used for OCR1 only, I believe.

How are bootstrappers configured for OCR2 then? Using @RensR's generic OCR2 bootstrap jobspec?

Reply:
I have no idea. Will ask around.

Reply:
Yes, for OCR2 we use the bootstrap job spec and don't do bootstrappers within OCR2 itself

Reply:
Just examined the PR, this code is needed. The `boostrap` spec relies on OCR2 to function and requests a `NewOCR2Provider` for its internal functionality.